### PR TITLE
Moved config.ini.example to config.defaults.ini

### LINKED ISSUE #63: crash on missing config option
If a config option is missing (e.g. commented out) the terminal is crashing when it searches the config for it.

### PR DESCRIPTION
**Pull request by sedrubal**
- see #43
- can fix #63 

I think later we should change the `run.py` parameters. I added a TODO in the code:

``` python
# TODO restructure this: simply provide a --config-file parameter but
# always load config.defaults.ini first and the the users config
```

### CODE REVIEW

**Comment by mgmax** [comment]
Renaming the config without changing the way the config system works is confusing, because the new name is just wrong. This should not be merged yet.

**Comment by sedrubal** [comment]
I changed the system by loading `config.defaults.ini` before loading `config.ini`, but I forgot to mention this in the commit message. See https://github.com/fau-fablab/FabLabKasse/commit/8a5a6373b6e752b13cd9411178128ac4cdbb03b1#diff-22cd404ff64283f596d2fe30d447f59dR96

(I just force pushed, because the PR was not clean. It contained changes of another PR)

File: TODO [hunk 1]
```diff
@@ -4,7 +4,7 @@ TODO
 Needed testing for FAU FabLab
 -----------------------------
 
-    cp config.ini.example config.ini
+    cp config.defaults.ini config.ini
```
Comment (mgmax) [inline TODO:7]: well, this is not really true anymore...

File: run.py [hunk 2]
```diff
@@ -59,11 +61,10 @@ def main():
                 time.sleep(2)
         else:
             print("loading example configuration file. edit FabLabKasse/config.ini to change. You do not need the --example parameter later.")
-            shutil.copyfile("config.ini.example", "config.ini")
+            shutil.copyfile("config.defaults.ini", "config.ini")
```
Comment (mgmax) [inline run.py:64]: same here

**Comment by sedrubal** [comment]
This PR does not completely fix the broken config system, but it might be a step. The milestone [ImproveConfig](https://github.com/fau-fablab/FabLabKasse/milestones/ImproveConfig) might help to keep in mind that further changes are required.

**Comment by patkan** [comment]
But doesn't this code simply load `config.defaults.ini` and then discard these settings and load `config.ini` if the latter file is present?
Or do these settings "overload" the old settings (thus adding only the changed parameters).

**Comment by sedrubal** [comment]
As far as I read and tested, the second reading overloads the settings of the first reading.

Example:
- `config.defaults.ini` contains `support_mail` but `config.ini` doesn't, the value set in defaults will be taken
- `config.defaults.ini` contains `support_mail = change@me` and `config.ini` contains `support_mail = kasse@fablab.fau.de` the latter one will be accessible

**Comment by sedrubal** [comment]
Oh, I think we want to do it like this example: https://docs.python.org/2/library/configparser.html#ConfigParser.RawConfigParser.read (but maybe we want to check if `config.ini` exists)
#rtfm

**Comment by mgmax** [comment]
From the source it looks quite good, but the documentation and handling of creating the first config needs to be changed. We don't really need a config anymore, maybe FabLabKasse should create some empty dummy config file with short instructions?

**Comment by mgmax** [comment]
closing as WONTFIX

**Pull request closed without merging**